src/compass/utils/runconfig.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from itertools import cycle
+import os
+from types import SimpleNamespace
+
+import journal
+import yamale
+from ruamel.yaml import YAML
+
+from compass.utils import helpers
+from compass.utils.wrap_namespace import wrap_namespace
+from sentinel1_reader.sentinel1_burst_slc import Sentinel1BurstSlc
+from sentinel1_reader.sentinel1_orbit_reader import get_swath_orbit_file_from_list
+from sentinel1_reader.sentinel1_reader import burst_from_zip
+
+
+def validate_group_dict(group_cfg: dict) -> None:
+    """Check and validate runconfig entries.
+
+    Parameters
+    ----------
+    group_cfg : dict
+        Dictionary storing runconfig options to validate
+    """
+    error_channel = journal.error('runconfig.validate_group_dict')
+
+    # Check 'input_file_group' section of runconfig
+    input_group = group_cfg['input_file_group']
+    # If is_reference flag is False, check that file path to reference
+    # burst is assigned and valid (required by geo2rdr and resample)
+    is_reference = input_group['reference_burst']['is_reference']
+    if not is_reference:
+        helpers.check_file_path(input_group['reference_burst']['file_path'])
+
+    # Check SAFE files
+    run_pol_mode = group_cfg['processing']['polarization']
+    for safe_file in input_group['safe_file_path']:
+        # Check if files exists
+        helpers.check_file_path(safe_file)
+
+        # Check safe pol mode
+        helpers.check_file_polarization_mode(safe_file, run_pol_mode)
+
+    for orbit_file in input_group['orbit_file_path']:
+        helpers.check_file_path(orbit_file)
+
+    # Check 'dynamic_ancillary_file_groups' section of runconfig
+    # Check that DEM file exists and is GDAL-compatible
+    dem_path = group_cfg['dynamic_ancillary_file_group']['dem_file']
+    helpers.check_file_path(dem_path)
+    helpers.check_dem(dem_path)
+
+    # Check 'product_path_group' section of runconfig.
+    # Check that directories herein have writing permissions
+    product_path_group = group_cfg['product_path_group']
+    helpers.check_write_dir(product_path_group['product_path'])
+    helpers.check_write_dir(product_path_group['scratch_path'])
+    helpers.check_write_dir(product_path_group['sas_output_file'])
+
+
+def load_bursts(cfg: SimpleNamespace) -> list[Sentinel1BurstSlc]:
+    '''For each burst find corresponding orbit'
+
+    Parameters
+    ----------
+    cfg : SimpleNamespace
+        Configuration of bursts to be loaded.
+
+    Returns
+    -------
+    _ : list[Sentinel1BurstSlc]
+        List of bursts loaded according to given configuration.
+    '''
+    error_channel = journal.error('runconfig.correlate_burst_to_orbit')
+
+    # dict to store bursts keyed by burst_ids
+    bursts = {}
+
+    # zip pol and IW subswath indices together
+    mode_to_pols = {'co-pol':['VV'], 'cross-pol':['VH'], 'dual-pol':['VV', 'VH']}
```

Comment:
why we are missing HH, HV and {HH, HV} ?
I'm sorry probably my many comments about polarization have complicated this part! But I think we are almost there. I very much liked that in the schema you removed HH, VV, VH and HV and replaced with 'co-pol' 'cross-pol' and 'dual-pol'. However, that does not mean that we should completely ignore HH and HV. 
With the current design of the schema for polarization (which I think is the right one), we need the following logic here:

```
mode_to_pols = check_file_polarization_mode(filename) 
 ```
basically we should not hardcode mode_to_pols but rather construct it based on the polarization in filename which would have these options in your check_file_polarization_mode function:
```
if pol == SV:
    mode_to_pols = {'co-pol':['VV']}
elif pol == DV:
    mode_to_pols = {'co-pol':['VV'], 'cross-pol':['VH'], 'dual-pol':['VV', 'VH']}
elif pol == SH:
    mode_to_pols = {'co-pol':['HH']}
else:
    mode_to_pols = {'co-pol':['HH'], 'cross-pol':['HV'], 'dual-pol':['HH', 'HV']}
```

Reply:
Addressed with 2b53b51